# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../tcp_wrappers/0016-tcp_wrappers-7.6-fix-warnings.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -ruNp tcp_wrappers_7.6.orig/fix_options.c tcp_wrappers_7.6/fix_options.c
--- tcp_wrappers_7.6.orig/fix_options.c	2006-03-01 23:45:28.000000000 +0100
+++ tcp_wrappers_7.6/fix_options.c	2006-03-01 23:45:25.000000000 +0100
@@ -50,7 +50,7 @@ struct request_info *request;
     struct in_addr dummy;
 #ifdef INET6
     struct sockaddr_storage ss;
-    int sslen;
+    socklen_t sslen;
 
     /*
      * check if this is AF_INET socket
diff -ruNp tcp_wrappers_7.6.orig/options.c tcp_wrappers_7.6/options.c
--- tcp_wrappers_7.6.orig/options.c	2006-03-01 23:45:28.000000000 +0100
+++ tcp_wrappers_7.6/options.c	2006-03-01 22:55:44.000000000 +0100
@@ -41,6 +41,7 @@ static char sccsid[] = "@(#) options.c 1
 #include <netinet/in.h>
 #include <netdb.h>
 #include <stdio.h>
+#include <unistd.h>
 #include <syslog.h>
 #include <pwd.h>
 #include <grp.h>
diff -ruNp tcp_wrappers_7.6.orig/scaffold.c tcp_wrappers_7.6/scaffold.c
--- tcp_wrappers_7.6.orig/scaffold.c	2006-03-01 23:45:28.000000000 +0100
+++ tcp_wrappers_7.6/scaffold.c	2006-03-01 22:56:13.000000000 +0100
@@ -17,6 +17,7 @@ static char sccs_id[] = "@(#) scaffold.c
 #include <arpa/inet.h>
 #include <netdb.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <syslog.h>
 #include <setjmp.h>
 #include <string.h>
